sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -292,6 +297,17 @@ export class ApplicationService extends RecordDataModelService<Application> {
       const applicationRepository =
         transactionalEntityManager.getRepository(Application);
       await applicationRepository.save(newApplication);
+
+      // Check if the student submitted E2 restriction
```

Comment:
The business context in the comment can be updated as `Check if the application requires E2 restriction check.`